### PR TITLE
move from ApprovalTests to Verify for api tests

### PR DESCRIPTION
**Pull request by SimonCropp**
* removes DiffPlex code since it comes pre-packaged https://github.com/VerifyTests/Verify.DiffPlex
 * removes `UseReporter` code. Difftool is now picked as per https://github.com/VerifyTests/DiffEngine/blob/main/docs/diff-tool.order.md and can be configured at the machine level

### CODE REVIEW

File: src/core/Akka.API.Tests/CoreAPISpec.cs [hunk 1]
```diff
@@ -5,279 +5,121 @@
...
     public class CoreAPISpec
     {
-        [Fact]
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        public void ApproveCore()
+        static CoreAPISpec()
         {
-            var publicApi = Filter(GeneratePublicApi(typeof(ActorSystem).Assembly));
-            Approvals.Verify(publicApi);
+            VerifierSettings.ScrubLinesContaining("[assembly: ReleaseDateAttribute(");
```
Comment (Aaronontheweb) [inline src/core/Akka.API.Tests/CoreAPISpec.cs:32]: Very cool

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Comment by Aaronontheweb** [comment]
Thank you very much for your contribution @SimonCropp !

File: src/core/Akka.API.Tests/CoreAPISpec.cs (hunk 1 above)
Comment (SimonCropp) [reply src/core/Akka.API.Tests/CoreAPISpec.cs:32]: yeah the scrubbers are pretty extensive https://github.com/VerifyTests/Verify/blob/main/docs/scrubbers.md

**Comment by SimonCropp** [comment]
this might also help you https://github.com/VerifyTests/Verify/blob/main/docs/build-server.md

**Pull request merged**